CDTDatastoreTests/DatastoreManagerTests.m

```diff
@@ -96,6 +97,40 @@ -(void) testSchema6ToSchema100Upgrade {
     }
 }
 
+- (void) testDatastoreClosesFilehandles {
+    // repeatedly obtain the same datastore in a loop, adding documents to it and calling ensureIndexed
+    // on each iteration, check that we have not excessively leaked filehandles, which would  indicate that the datastore and indexmanager are not being dealloc'd correctly
+    int n = 1000;
+    CDTDatastore *ds;
+    for (int i=0; i<n; i++) {
+        @autoreleasepool {
+            CDTDocumentRevision *rev = [CDTDocumentRevision revision];
+            rev.body = [NSMutableDictionary dictionaryWithDictionary: @{@"hello": @"world"}];
+            NSError *err;
+            ds = [self.factory datastoreNamed:@"test" error:&err];
+            XCTAssertNil(err);
+            [ds createDocumentFromRevision:rev error:&err];
+            XCTAssertNil(err);
+            [ds ensureIndexed:@[@"hello"] withName:@"index"];
+            // if the datastore has been properly released, we will not run out of filehandles
+            int pid = [[NSProcessInfo processInfo] processIdentifier];
+            // first we call with a null pointer to see what the minimum buffer size needed is
+            int bufferSize = proc_pidinfo(pid, PROC_PIDLISTFDS, 0, 0, 0);
+            // buffer size is bytes, we want number of processes
+            struct proc_fdinfo *fdInfo = (struct proc_fdinfo *)malloc(bufferSize);
+            int fdCount = bufferSize / PROC_PIDLISTFD_SIZE;
+            // now call with the buffer to get the actual number of processes (may be lower than buffer size)
+            bufferSize = proc_pidinfo(pid, PROC_PIDLISTFDS, 0, fdInfo, fdCount);
+            // buffer size is bytes, we want number of processes
+            fdCount = bufferSize / PROC_PIDLISTFD_SIZE;
+            // as observed in testing, this should never go above 8, but we'll set a conservative limit of 100 to allow some breathing room
+            XCTAssertTrue(fdCount < 100);
```

Comment:
Can we break the loop if this fails? Currently the test result repeats the failure many times, but I think one is enough for us to know it has gone wrong.